### ISSUE TITLE
Uncaught TypeError: event.preventDefault is not a function

### ISSUE THREAD

**Issue by pleonex** [amitab/atom-cscope#11]
1. Just after toggling CScope plugin. Could be because I haven't created the cscope.out yet? (I thought this is done automatically) **EDIT:** I have create the DB and still happens each time I toggle the plugin.

**Atom Version**: 1.6.0-beta0-1b416c7
**System**: "Fedora release 23
**Thrown From**: [atom-cscope](https://github.com/amitab/atom-cscope) package, v0.13.0
### Stack Trace

Uncaught TypeError: event.preventDefault is not a function

```
At /home/benito/.atom/packages/atom-cscope/lib/views/input-view.coffee:41

TypeError: event.preventDefault is not a function
    at InputView.module.exports.InputView.searchCallback (/home/benito/.atom/packages/atom-cscope/lib/views/input-view.coffee:41:11)
    at /home/benito/.atom/packages/atom-cscope/lib/views/input-view.coffee:1:1
    at Function.module.exports.Emitter.simpleDispatch (/usr/local/share/atom/resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/usr/local/share/atom/resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at /usr/local/share/atom/resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:1327:25
```
### Commands

```
     -4:41.3.0 settings-view:open (atom-text-editor.editor.is-focused)
     -4:37.4.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:52.4.0 find-and-replace:show (atom-text-editor.editor.is-focused)
     -0:51.6.0 core:paste (atom-text-editor.editor.mini.is-focused)
  4x -0:50.8.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:07.5.0 editor:consolidate-selections (atom-text-editor.editor.is-focused)
     -0:07.5.0 git-diff-details:close-git-diff-details (atom-text-editor.editor.is-focused)
     -0:07.5.0 core:cancel (atom-text-editor.editor.is-focused)
     -0:06.1.0 command-palette:toggle (atom-text-editor.editor.is-focused)
     -0:01.7.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:01.7.0 atom-cscope:toggle (atom-text-editor.editor)
```
### Config

``` json
{
  "core": {
    "autoHideMenuBar": true,
    "disabledPackages": [
      "rti-linter-code",
      "latex"
    ],
    "themes": [
      "atom-material-ui",
      "one-dark-syntax"
    ]
  },
  "atom-cscope": {}
}
```
### Installed Packages

``` coffee
# User
atom-cscope, v0.13.0
atom-material-ui, v0.8.1
autocomplete-clang, v0.8.9
autocomplete-python, v1.6.1
autocomplete-xml, v0.8.1
docblockr, v0.7.3
file-icons, v1.6.14
git-diff-details, v0.21.0
git-log, v0.4.1
hyperclick, v0.0.35
language-latex, v0.6.1
linter, v1.11.3
linter-alex, v2.0.0
linter-chktex, v1.2.5
linter-clang, v3.4.1
linter-pep8, v1.2.1
linter-xmllint, v1.3.1
minimap, v4.19.0
minimap-bookmarks, v0.2.0
minimap-cursorline, v0.1.0
minimap-find-and-replace, v4.4.0
minimap-git-diff, v4.1.8
minimap-highlight-selected, v4.3.1
minimap-linter, v1.1.1
minimap-selection, v4.3.1
pdf-view, v0.39.0
term3, v0.21.2
tool-bar, v0.2.0
tool-bar-almighty, v0.6.1

# Dev
No dev packages
```

**Comment by amitab** [comment]
Hmm... Seems odd. `event.preventDefault()` is a function available to javascript, not something specific to `atom-cscope`.
Also I'm really sorry, but I'm not able to reproduce this error. But I would like to find out what exactly is causing this. Please make the following changes:

In a new line after `/home/benito/.atom/packages/atom-cscope/lib/views/input-view.coffee:41`

``` diff
    event.preventDefault() if event
+   console.log event
    false
```
1. Reload atom editor
2. Press `ctrl-shift-i` to open up developer tools and select the console tab
3. Now do the same sequence which previously caused the error

You should get something like this in the console:

`j…y.Event {originalEvent: CustomEvent, type: "core:confirm", timeStamp: 1455297943467, jQuery2140058944008545950055: true, which: undefined…} input-view.coffee:42`

Expand this object and you will get a list of attributes.
Then expand the `__proto__` attribute too.

Probably the `preventDefault` attribute isn't there. Which is why you must be getting this error. 

This is how my example looks like:
![Test Image](http://i.imgur.com/lGu7TUh.png)

Let me know what you find out.

**Comment by pleonex** [comment]
`event` is just empty (`[]`). This is the callstack with the inner method that trigger this method: 
![callstack](https://cloud.githubusercontent.com/assets/3107481/13016048/b6c95aa0-d1bc-11e5-8bd6-3f763e69b125.png)

**EDIT:** I tried to remove the _space-pen_ package dependency because it seems it's decrapted ([although there is no alternative](https://github.com/atom/atom-space-pen-views/issues/16)) an only use _atom-space-pen-views_ package but still its happening.

**Comment by amitab** [comment]
Oh this was just an issue with atom 1.6.0-beta. Turns out, the TextBuffer now sends the actual changes in the buffer instead of `undefined`.
This is not really a big deal. For now, you can simply make the following changes:

``` diff
    @parentView.removeLoading()
-   event.preventDefault() if event
+   event.preventDefault() if event.preventDefault
    false
```

I will make the fix and send it out for the next release.

**Comment by pleonex** [comment]
Thanks!

I had  to change it to `event.preventDefault() if event && event.preventDefault` because it was `undefined` sometimes too.

**Comment by amitab** [comment]
Fixed in master c7f3ebbafad2d04c9e0c87557e165a75afca4306
